devicetypes/smartthings/aeotec-doorbell-6.src/aeotec-doorbell-6.groovy

```diff
@@ -0,0 +1,225 @@
+				attributeState "both", label: 'ring!', action: 'alarm.off', icon: "st.alarm.alarm.alarm", backgroundColor: "#0e7507"
+			}
+		}
+		standardTile("off", "device.alarm", inactiveLabel: false, decoration: "flat") {
+			state "default", label: '', action: "alarm.off", icon: "st.secondary.off"
+		}
+		standardTile("reset", "device.reset", height: 2, width: 2, inactiveLabel: false, decoration: "flat") {
+			state "default", label: "reset", action: "reset"
+		}
+		valueTile("tamper", "device.tamper", height: 2, width: 2, decoration: "flat") {
+			state "clear", label: 'tamper clear', backgroundColor: "#ffffff"
+			state "detected", label: 'tampered', backgroundColor: "#ffffff"
+		}
+
+		main "alarm"
+		details(["alarm", "off", "tamper"])
+	}
+}
+
+private getNumberOfSounds() {
+	def numberOfSounds = ["0003" : 8]
+	return numberOfSounds[zwaveInfo.prod] ?: 1
+}
+
+def installed() {
+	initialize()
+	sendEvent(name: "alarm", value: "off", isStateChange: true)
+}
+
+def updated() {
+	initialize()
+}
+
+def initialize() {
+	if (!childDevices) {
+		addChildren(numberOfSounds)
+	}
+	state.soundTable = []
+	for (def i in 1..numberOfSounds) {
+		state.soundTable[i] = false
+	}
+	sendEvent(name: "checkInterval", value: 2 * 60 * 60 + 2 * 60, displayed: false, data: [protocol: "zwave", hubHardwareId: device.hub.hardwareID])
+}
+
+def parse(String description) {
+	def result = null
+	def cmd = zwave.parse(description)
+	if (cmd) {
+		result = zwaveEvent(cmd)
+	}
+	log.debug "Parse returned: ${result.inspect()}"
+	return result
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.basicv1.BasicReport cmd) {
+	createEvent(name: "alarm", value: cmd.value ? "both" : "off")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.multichannelv3.MultiChannelCmdEncap cmd) {
+	def encapsulatedCommand = cmd.encapsulatedCommand()
+	def endpoint = cmd.sourceEndPoint
+	if (endpoint == 1) {
+		zwaveEvent(encapsulatedCommand)
+	} else {
+		String childDni = "${device.deviceNetworkId}:$endpoint"
+		def child = childDevices.find { it.deviceNetworkId == childDni }
+	}
+}
+
+def on() {
+	setActiveSound(1)
+	def cmds = []
+	cmds << zwave.basicV1.basicSet(value: 0xFF).format()
+	cmds
+}
+
+def off() {
+	setActiveSound(1)
+	def cmds = []
+	cmds << zwave.basicV1.basicSet(value: 0x00).format()
+	resetActiveSounds()
+	cmds
+}
+
+def strobe() {
+	on()
+}
+
+def siren() {
+	on()
+}
+
+def both() {
+	on()
+}
+
+def ping() {
+	zwave.basicV1.basicGet()
+}
+
+private addChildren(numberOfSounds) {
+	for (def endpoint : 2..numberOfSounds) {
+		try {
+			String childDni = "${device.deviceNetworkId}:$endpoint"
+			def componentLabel = device.displayName[0..-2] + "$endpoint"
+
+			addChildDevice("Aeotec Child Siren", childDni, device.getHub().getId(), [
+					completedSetup: true,
+					label         : componentLabel,
+					isComponent   : false,
+					componentName : "Aeotec Doorbell 6 $endpoint",
+					componentLabel: "Aeotec Doorbell 6 $endpoint"
```

Comment:
I think we should be making this an MCD (`isComponent: true`) but if we're not, you must remove the `componentName` and `componentLabel` arguments.

@dkirker where did we come down on MCD component names/labels?

Reply:
Tom wrote some guidelines in Confluence that I've linked to in the ticket.

Reply:
Please see the guidelines tom has written for multi-component devices that I linked in the Jira ticket. The component name should probably be something like `sound1` and the componentLabel should probably be `Sound 1`

Reply:
Or, if there is a more descriptive name for the sound, then it could probably be that -- much like what I did with the Ikea Button. I suspect at some point these will get some sort of translation.

Reply:
this still has not been addressed